### PR TITLE
Feature - Copy url of a file to clipboard

### PR DESCRIPTION
**Pull request by kasvith**
Pull Request for Issue #430  .

### Summary of Changes
- Added a new button to the action list
- Added ability to copy the public url of a file to the clipboard with vanillaJS
- Added localizable strings for new feature (Default texts can be changed in language files)

### Testing Instructions
- Open NMM
- Open Action list of a file as follows and click on the link icon
image
- Click on **Get sharable link** button
image
- After some time you will get the sharable url of the file as follows
image
- You can copy it using the button next to text box or copy it your own

### Expected result
- Url of the file should be in your clipboard
- Try pasting it out

### Actual result
- It works 💯 

### Documentation Changes Required
- [document.execCommand('copy')](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand) is supported by IE9+ and others, The text box is there if browser does not support copy API
- I've not added any accessibility setups

### CODE REVIEW

**Comment by dneukirchen** [comment]
Thank you kasun. It looks good, but for me it feels a little weird to get a sharable link for a static file on local filesystem. Imo this is another feature where the cloud plugins overcomplicate things. 

UX/UI is a matter of taste. Like posted in the original issue, i would prefer to have the copy/paste input in the infobar on the right and get rid of modals.

**Comment by kasvith** [comment]
@dneukirchen yeah i would like that too because in the list view we will not have this feature.

basically this should be same for every adapter, make them more generalized, sharing a local file also would be nice if we adding ACL features to certain files. 

For cloud and other adapters this is preferred.

**Comment by kasvith** [comment]
@laoneo what's your idea on that?

**Comment by laoneo** [comment]
Please revert the change in the banner.jpg. If @dneukirchen has approved it, I will merge it.

**Comment by kasvith** [comment]
@laoneo done, did not notice that lol

**Comment by laoneo** [comment]
For a second part (doesn't need to be done in this pr), we can skip the get shareable link step for local files.

**Comment by kasvith** [comment]
hmm...but i still prefer to have it as we don't get full url here by default. It's generated by the API

**Comment by laoneo** [comment]
That's why it is better to do it in a second pr, to not block this one.

**Comment by kasvith** [comment]
Yeah we have the basic functionality in here, then its just adjusting it

**Comment by laoneo** [comment]
Conflicts again

**Comment by kasvith** [comment]
Yeah i will rebuild scripts as Search box was merged

**Comment by kasvith** [comment]
Resolved conflicts

File: administrator/language/en-GB/en-GB.com_media.ini [hunk 1]
```diff
@@ -67,3 +68,6 @@ COM_MEDIA_TOGGLE_INFO="Toggle item info"
 COM_MEDIA_TOGGLE_LIST_VIEW="Toggle between grid and table views"
 COM_MEDIA_TOGGLE_SELECT_ITEM="Toggle select item"
 COM_MEDIA_UPDLOAD_SUCCESS="Item uploaded."
+COM_MEDIA_SHARE="Share the file"
```
Comment (laoneo) [inline administrator/language/en-GB/en-GB.com_media.ini]: We have the policy in Joomla to alpha order the strings.

Comment (kasvith) [reply administrator/language/en-GB/en-GB.com_media.ini]: Will fix asap

**Comment by kasvith** [comment]
@laoneo done it

**Comment by laoneo** [comment]
Very good job, I see we have a new vue.js master in our team!

**Comment by kasvith** [comment]
Still learning from a master @dneukirchen

**Pull request merged**